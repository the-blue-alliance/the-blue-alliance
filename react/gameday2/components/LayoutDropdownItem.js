import React, { PropTypes } from 'react'

var LayoutDropdownItem = React.createClass({
  propTypes: {
    layoutId: PropTypes.number.isRequired,
    handleClick: PropTypes.func.isRequired
  },
  handleClick: function(event) {
    if (this.props.handleClick) {
      event.preventDefault()
      this.props.handleClick(this.props.layoutId)
    }
  },
  render: function() {
    return (
      <li onClick={this.handleClick}><a href='#'>{this.props.children}</a></li>
    )
  },
})

export default LayoutDropdownItem
