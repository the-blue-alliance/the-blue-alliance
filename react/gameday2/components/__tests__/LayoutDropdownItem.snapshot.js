import React from 'react'
import renderer from 'react-test-renderer'
import LayoutDropdownItem from '../LayoutDropdownItem'

describe('LayoutDropdownItem snapshot', () => {
  it('renders correctly', () => {

    const item = renderer.create(
      <LayoutDropdownItem layoutId={0} handleClick={() => {}}>Test Item</LayoutDropdownItem>
    ).toJSON()
    expect(item).toMatchSnapshot()
  })
})
