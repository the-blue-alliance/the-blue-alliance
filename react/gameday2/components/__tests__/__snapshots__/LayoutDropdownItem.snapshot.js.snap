exports[`LayoutDropdownItem snapshot renders correctly 1`] = `
<li
  onClick={[Function bound handleClick]}>
  <a
    href="#">
    Test Item
  </a>
</li>
`;
